lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -33,9 +34,33 @@ def register
 
     # Receive an array of events and immediately attempt to index them (no buffering)
     def multi_receive(events)
+      until @template_installed.true?
+        sleep 1
+      end
       retrying_submit(events.map {|e| event_action_tuple(e)})
     end
 
+    def after_successful_connection
+      @template_installer = Thread.new do
+          sleep_interval = @retry_initial_interval
+          until successful_connection? || @stopping.true?
+            @logger.debug("Waiting for connectivity to Elasticsearch cluster")
+            Stud.stoppable_sleep(sleep_interval) { @stopping.true? }
+            sleep_interval = next_sleep_interval(sleep_interval)
+          end
+          yield if successful_connection?
+        end
+      end
+
+    def stop_template_installer
+      @template_installer.join if @template_installer
+    end
+
+    def successful_connection?
+      # !!maximum_seen_major_version
```

Comment:
this leftover commented code can be removed
also, any particular reasoning to (not) use !! here?

Reply:
!! makes more sense in the context of a `?` method, so I'll use that